### ISSUE TITLE
Raspberry Pi Support?

### ISSUE THREAD

**Issue by dwaq** [pkourany/clickButton#15]
I've long used this library on my Particle boards, but recently moved my project to a Raspberry Pi. The RPi required some changes to the library to work successfully. Now that the Particle IDE is also targeting the RPi, I'm curious if you'd like to work together to support both the Particle boards and the RPi in this library?

I have some changes complete but will need to do some more testing to ensure it's 100% compatible with both product families.

**Comment by pkourany** [comment]
@dwaq, absolutely!  Put together a PR with your proposed changes and we can go from there.
